src/Model.elm

```diff
@@ -182,47 +191,140 @@ processFollowEvent relationship flag model =
-
-        UpdateVisibility visibility ->
-            { draft | visibility = visibility } ! []
-
-        UpdateReplyTo status ->
-            { draft
-                | in_reply_to = Just status
-                , status = Mastodon.Helper.getReplyPrefix currentUser status
-                , sensitive = Maybe.withDefault False status.sensitive
-                , spoiler_text =
-                    if status.spoiler_text == "" then
-                        Nothing
-                    else
-                        Just status.spoiler_text
-                , visibility = status.visibility
-            }
-                ! [ Command.focusId "status" ]
+updateDraft : DraftMsg -> Account -> Model -> ( Model, Cmd Msg )
+updateDraft draftMsg currentUser model =
+    let
+        draft =
+            model.draft
+    in
+        case draftMsg of
+            ClearDraft ->
+                { model | draft = defaultDraft } ! []
+
+            ToggleSpoiler enabled ->
+                let
+                    newDraft =
+                        { draft
+                            | spoiler_text =
+                                if enabled then
+                                    Just ""
+                                else
+                                    Nothing
+                        }
+                in
+                    { model | draft = newDraft } ! []
+
+            UpdateSensitive sensitive ->
+                { model | draft = { draft | sensitive = sensitive } } ! []
+
+            UpdateSpoiler spoiler_text ->
+                { model | draft = { draft | spoiler_text = Just spoiler_text } } ! []
+
+            UpdateVisibility visibility ->
+                { model | draft = { draft | visibility = visibility } } ! []
+
+            UpdateReplyTo status ->
+                { model
+                    | draft =
+                        { draft
+                            | in_reply_to = Just status
+                            , status = Mastodon.Helper.getReplyPrefix currentUser status
+                            , sensitive = Maybe.withDefault False status.sensitive
+                            , spoiler_text =
+                                if status.spoiler_text == "" then
+                                    Nothing
+                                else
+                                    Just status.spoiler_text
+                            , visibility = status.visibility
+                        }
+                }
+                    ! [ Command.focusId "status" ]
+
+            UpdateInputInformation { status, selectionStart } ->
+                let
+                    newModel =
+                        { model
+                            | draft = { draft | status = status }
+                            , autoCursorPosition = selectionStart
+                        }
+
+                    stringToPos =
+                        (String.slice 0 newModel.autoCursorPosition status)
```

Comment:
Nit: unnecessary wrapping parenthesis (here and in many other places in this patch)